app/code/Magento/SalesRule/Model/Rule/Condition/Product.php

```diff
@@ -26,20 +28,47 @@ protected function _addSpecialAttributes(array &$attributes)
         $attributes['quote_item_row_total'] = __('Row total in cart');
     }
 
+    /**
+     * @param \Magento\Framework\Model\AbstractModel $model
+     *
+     * @return \Magento\Catalog\Api\Data\ProductInterface|\Magento\Catalog\Model\Product
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    protected function getProductToValidate(\Magento\Framework\Model\AbstractModel $model)
```

Comment:
Please, make this method private

Reply:
Done, please review: fae98c01e29ea9ca47eca00cf2081ede1f729ccb & 9a35b459f1b31b2b8d40c6597bd90ac9c5c865fe